### PR TITLE
Login UI Revamp

### PR DESCRIPTION
**Pull request by SmitGala**
## Summary by CodeRabbit

- **New Features**
  - Introduced new user login components for various authentication methods including MFA, SSO, and username/password.
  - Added a "Recover Password" feature with user interface components for password recovery and reset.
  - Implemented a new component for custom action link placement in text fields for enhanced customization.
  - Added Single Sign-On (SSO) login functionality and interface.
  
- **Enhancements**
  - Updated styling across various components to improve layout and user experience.
  - Enhanced error handling and messaging for login processes.
  - Improved flexibility in text field customization through new style definitions.
  
- **Bug Fixes**
  - Corrected text and translations in various components to enhance clarity and user interaction.
  
- **Documentation**
  - Updated component usage in templates to reflect new naming and structural changes.
  
- **Tests**
  - Extended testing suites to cover new components and functionalities, including integration tests for password recovery.

### CODE REVIEW

File: app/components/user-login/perform-mfa/index.hbs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      @placeholder={{t 'enterCode'}}
+      @labelTypographyColor='textPrimary'
+      @labelTypographyVariant='body2'
+      @textFieldRootClass={{@textFieldClass}}
+      @value={{@otp}}
+      @autofocus={{true}}
+      autocomplete='one-time-code'
+      data-test-user-login-perform-mfa-otp-input
+    />
+    <AkButton
```
Comment (future-pirate-king) [inline app/components/user-login/perform-mfa/index.hbs:42]: add new line above

File: app/components/user-login/recover-password/index.hbs [hunk 2]
```diff
@@ -0,0 +1,97 @@
+<AkStack
+  @alignItems='center'
+  @justifyContent='center'
+  local-class='inner-container'
```
Comment (future-pirate-king) [inline app/components/user-login/recover-password/index.hbs:4]: can we rename this to `recover-password-root`

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/components/user-login/index.scss [hunk 3]
```diff
@@ -0,0 +1,28 @@
+.inner-container {
+  height: 100vh;
+
+  .login-container {
+    border: 1px solid #dddddd;
+    border-radius: 3px;
+    min-width: 29.5em;
+    max-width: 29.5em;
+    box-shadow: 5px 5px 30px 10px #0000000d;
```
[the remark this answers is not in the document]
Comment (SmitGala) [reply app/components/user-login/index.scss:9]: Have asked Sujith to tell me box-shadow from which we use

File: app/components/user-login/reset-password/index.scss [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  margin-top: 4px;
+
+  .error-icon {
+    padding-top: 2px;
+    font-size: 1em !important;
+  }
+
+  .error-message {
+    padding-left: 10px;
+    font-size: 0.857em;
```
Comment (future-pirate-king) [inline app/components/user-login/reset-password/index.scss:40]: use `rem` for font-size

File: app/components/user-login/via-sso/index.hbs [hunk 5]
```diff
@@ -0,0 +1,59 @@
+{{#if @isEnforced}}
+  <form ...attributes onsubmit='return false' class='w-full'>
```
Comment (future-pirate-king) [inline app/components/user-login/via-sso/index.hbs:2]: use `on` modifier for submit and remove click from button

File: app/components/user-login/via-sso/index.hbs [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    @showCredError={{@showCredError}}
+    @showAccountLockError={{@showAccountLockError}}
+    data-test-user-login-via-sso-not-forced-login
+  />
+
+  <div data-test-user-login-via-sso-or local-class='login-form-sso-separator'>
+    <div local-class='login-form-sso-separator-or'>
+      {{t 'or'}}
+    </div>
+  </div>
```
Comment (future-pirate-king) [inline app/components/user-login/via-sso/index.hbs:49]: this can be done with `ak-stack`, `ak-divider` & `ak-typography` without requiring additional css

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/components/user-login/via-username-password/index.hbs [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+      {{#if @showCredError}}
+        <AkStack local-class='error-stack'>
+          <AkIcon @iconName='cancel' @color='error' local-class='error-icon' />
+
+          <AkTypography @color='error' local-class='error-message'>
+            {{t 'credentialsIncorrect'}}
+          </AkTypography>
+        </AkStack>
+      {{/if}}
+
```
Comment (future-pirate-king) [inline app/components/user-login/via-username-password/index.hbs:76]: remove new line

File: app/components/user-login/via-username-password/index.scss [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+.error-stack {
+  margin-top: 4px;
+
+  .error-icon {
+    padding-top: 2px;
+    font-size: 1em !important;
+  }
+
+  .error-message {
+    padding-left: 10px;
```
Comment (future-pirate-king) [inline app/components/user-login/via-username-password/index.scss:15]: use em for padding and margin and rem for font-size

File: app/templates/login.hbs [hunk 9]
```diff
@@ -1,2 +1,2 @@
-{{page-title "Login"}}
-<LoginComponent></LoginComponent>
+{{page-title 'Login'}}
+<UserLogin />
```
Comment (future-pirate-king) [inline app/templates/login.hbs:2]: add new line above

File: cypress/support/Actions/auth/LoginActions.ts [hunk 10]
```diff
@@ -32,12 +32,12 @@ export default class LoginActions {
     // Wait for frontend config request to resolve
     cy.wait('@frontendConfig');
 
-    cy.findByPlaceholderText('Username / Email').type(username); // Username/Email field
-    cy.findByLabelText('login-user-check-icon').click();
+    cy.findByPlaceholderText('Enter your Username/Email ID').type(username); // Username/Email field
```
Comment (future-pirate-king) [inline cypress/support/Actions/auth/LoginActions.ts:35]: this should be coming through translations

File: cypress/support/Actions/auth/LoginActions.ts [hunk 11]
```diff
@@ -32,12 +32,12 @@ export default class LoginActions {
...
+    cy.findByLabelText('login-next-button').click();
 
     cy.wait('@checkUserRoute', { timeout: 30000 });
 
-    cy.findByPlaceholderText('Password').type(password); // Password field
+    cy.findByPlaceholderText('Enter the Password').type(password); // Password field
```
Comment (future-pirate-king) [inline cypress/support/Actions/auth/LoginActions.ts:40]: this should be coming through translations

File: cypress/support/Actions/auth/LoginActions.ts [hunk 12]
```diff
@@ -65,14 +65,15 @@ export default class LoginActions {
...
-        cy.findByText(cyTranslate('register'))
-          .should('exist')
-          .should('have.attr', 'href', res.registration_link);
+        cy.findByText(cyTranslate('registerToday')).should('exist');
       }
     });
 
-    cy.findByPlaceholderText('Username / Email').should('be.visible'); // Username/Email field
-    cy.findAllByLabelText('login-user-check-icon').should('exist'); // User check button;
+    cy.findByPlaceholderText('Enter your Username/Email ID').should(
```
Comment (future-pirate-king) [inline cypress/support/Actions/auth/LoginActions.ts:72]: this should be coming through translations

File: cypress/support/Actions/auth/LoginActions.ts [hunk 13]
```diff
@@ -89,8 +90,10 @@ export default class LoginActions {
         cacheAcrossSpecs,
         validate: () => {
           // If these elements are displayed, token is expired
-          cy.findByPlaceholderText('Username / Email').should('not.exist'); // Username/Email field
-          cy.findByLabelText('login-user-check-icon').should('not.exist'); // User check button;s
+          cy.findByPlaceholderText('Enter your Username/Email ID').should(
```
Comment (future-pirate-king) [inline cypress/support/Actions/auth/LoginActions.ts:93]: this should be coming through translations

File: tests/integration/components/user-login/index-test.js [hunk 14]
```diff
@@ -69,25 +69,31 @@ module('Integration | Component | login component', function (hooks) {
...
+      .exists();
+    assert
+      .dom('[data-test-user-login-via-username-password-password-input]')
+      .exists();
+    assert
+      .dom('[data-test-user-login-via-username-password-login-button]')
+      .exists();
+    assert
+      .dom('[data-test-user-login-via-username-password-username-input]')
       .hasValue('appknoxusername');
```
Comment (future-pirate-king) [inline tests/integration/components/user-login/index-test.js:97]: add new line between multiline statements

File: tests/integration/components/user-login/index-test.js
Comment (future-pirate-king) [inline tests/integration/components/user-login/index-test.js:1]: add proper separation between statements

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/services/whitelabel.ts [hunk 15]
```diff
@@ -68,4 +68,8 @@ export default class WhitelabelService extends Service {
 
     return this.configuration.imageData.logo_on_darkbg;
   }
+
+  get show_contact_support() {
+    return window.location.href.includes('secure.appknox.com');
```
Comment (future-pirate-king) [inline app/services/whitelabel.ts:73]: use window from `@service('browser/window')`

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

**Pull request merged**